contracts/cw2981-royalties/src/msg.rs

```diff
@@ -24,16 +34,179 @@ pub enum Cw2981QueryMsg {
+    Approvals {
+        token_id: String,
+        include_expired: Option<bool>,
+    },
+    /// Return approval of a given operator for all tokens of an owner, error if not set
+    #[returns(OperatorResponse)]
+    Operator {
+        owner: String,
+        operator: String,
+        include_expired: Option<bool>,
+    },
+    /// List all operators that can access all of the owner's tokens
+    #[returns(OperatorsResponse)]
+    AllOperators {
+        owner: String,
+        /// unset or false will filter out expired items, you must set to true to see them
+        include_expired: Option<bool>,
+        start_after: Option<String>,
+        limit: Option<u32>,
+    },
+    /// Total number of tokens issued
+    #[returns(NumTokensResponse)]
+    NumTokens {},
+
+    #[returns(CollectionInfo)]
+    ContractInfo {},
+
+    /// With MetaData Extension.
+    /// Returns top-level metadata about the contract
+    #[returns(CollectionInfo)]
+    GetCollectionInfo {},
+
+    #[returns(Ownership<Addr>)]
+    Ownership {},
+
+    /// With MetaData Extension.
+    /// Returns metadata about one particular token, based on *ERC721 Metadata JSON Schema*
+    /// but directly from the contract
+    #[returns(NftInfoResponse<Extension>)]
+    NftInfo { token_id: String },
+    /// With MetaData Extension.
+    /// Returns the result of both `NftInfo` and `OwnerOf` as one query as an optimization
+    /// for clients
+    #[returns(AllNftInfoResponse<Extension>)]
+    AllNftInfo {
+        token_id: String,
+        /// unset or false will filter out expired approvals, you must set to true to see them
+        include_expired: Option<bool>,
+    },
+
+    /// With Enumerable extension.
+    /// Returns all tokens owned by the given address, [] if unset.
+    #[returns(TokensResponse)]
+    Tokens {
+        owner: String,
+        start_after: Option<String>,
+        limit: Option<u32>,
+    },
+    /// With Enumerable extension.
+    /// Requires pagination. Lists all token_ids controlled by the contract.
+    #[returns(TokensResponse)]
+    AllTokens {
+        start_after: Option<String>,
+        limit: Option<u32>,
+    },
+
+    /// Return the minter
+    #[returns(MinterResponse)]
+    Minter {},
+
+    #[returns(Option<String>)]
+    GetWithdrawAddress {},
+
+    // -- below queries, Extension and GetCollectionInfoExtension, are just dummies, since type annotations are required for
+    // -- TMetadataExtension and TCollectionInfoExtension, Error:
+    // -- "type annotations needed: cannot infer type for type parameter `TMetadataExtension` declared on the enum `Cw721QueryMsg`"
+    #[returns(())]
+    Extension { msg: Extension },
 }
 
-impl Default for Cw2981QueryMsg {
-    fn default() -> Self {
-        Cw2981QueryMsg::CheckRoyalties {}
+// impl Default for Cw2981QueryMsg {
+//     fn default() -> Self {
+//         Cw2981QueryMsg::CheckRoyalties {}
+//     }
+// }
+
+// impl CustomMsg for Cw2981QueryMsg {}
+
```

Comment:
```suggestion
```

Reply:
This was some test code from me, I removed it.